src/scraping/tools.py

```diff
@@ -253,3 +256,52 @@ def dataframe_to_db(self) -> None:
                     if_exists='replace',
                     index=False
                     )
+
+
+class GoogleSheets(DataBaseInteraction):
+    def __init__(self) -> None:
+        super().__init__()
+
+    @staticmethod
+    def base64_to_json(b64) -> dict[str, str]:
+        decodedbytes: bytes = base64.b64decode(b64[1:-1])
+        decodedstr: str = decodedbytes.decode("ascii")
+        json_dict: dict[str, str] = json.loads(decodedstr)
+        return json_dict
+
+    def read_from_sql(self, table_name: str) -> None:
+        self.db_connect()
+        df: pd.DataFrame = pd.read_sql(table_name, con=self.db_engine)
+        df['date_joined'] = df['date_joined'].dt.strftime('%Y-%m-%d')
+        df['date_link_sent'] = df['date_link_sent'].dt.strftime('%Y-%m-%d')
+        df.replace(np.nan, 'N/A', inplace=True)
+        self.dataframe: pd.DataFrame = df
+
+    def writetosheets(self) -> dict[str, Any]:
+        SCOPES: list[str] = ['https://www.googleapis.com/auth/spreadsheets']
+        b64: str = os.environ['SHEETS_API_CREDENTIALS_B64']
+        json_dict: dict[str, str] = self.base64_to_json(b64)
+
+        # create service account credentials object
+        creds = service_account.Credentials.from_service_account_info(json_dict, scopes=SCOPES)
+
+        # Construct a Resource for interacting with an API
+        service = build('sheets', 'v4', credentials=creds)
+
+        # convert dataframe into list, with first list being column names
+        data: list[list[Any]] = self.dataframe.to_numpy().tolist()
+        column_names: list[str] = self.dataframe.columns.tolist()
+        data.insert(0, column_names)
```

Comment:
maybe give a name to the 0, so we know what it is? put it in a variable?

Reply:
maybe you can put data = column_names (already as a list of lists with only 1 list) + data

Reply:
(maybe add this to the data pre-processing)?